props/saber_fett263_buttons.h

```diff
@@ -381,160 +1813,2794 @@ SaberFett263Buttons() : PropBase() {}
+               current_preset_.Save();
+               MenuSave();
+               return true;
+            case MENU_CLASH_LOCATION:
+            case MENU_DRAG_SIZE:
+            case MENU_EMITTER_SIZE:
+              SaveStyleSetting();
+              return true;
+            case MENU_DELETE:
+              if (choice_) {
+                current_preset_.SaveAt(-1);
+                choice_ = false;
+                menu_type_ = MENU_TOP;
+                MenuSelect();
+                return true;
+              }
+              PlayMenuSound("maffirm.wav");
+              return true;
+            case MENU_EXIT:
+              PlayMenuSound("mselect.wav");
+              menu_ = false;
+              menu_type_ = MENU_TOP;
+              wav_player.Free();
+              return true;
+#endif
+            default:
+              break;  
+          }
+        } else {
+          if (!swinging_) {
+            check_blast_ = false;
+            swing_blast_ = false;
 #ifdef FETT263_BM_DISABLE_OFF_BUTTON
-          if (!battle_mode_) {
+            if (!battle_mode_ && !saved_gesture_control.powerlock) {
+              Off();
+              saber_off_time_ = millis();
+            }
+#else
+            if (!saved_gesture_control.powerlock) {
             Off();
             saber_off_time_ = millis();
-          }
-#else
-          Off();
-          saber_off_time_ = millis();
-          battle_mode_ = false;
+            battle_mode_ = false;
+            }
 #endif
+#ifdef FETT263_SAVE_CHOREOGRAPHY
+            if (choreo_ && saved_choreography.clash_rec[clash_count_].stance == SavedRehearsal::STANCE_END) {
+              choreo_ = false;
+              if (SFX_clsh) {
+                SFX_clsh.Select(-1);
+              } else {
+                SFX_clash.Select(-1);
+              }
+              if (SFX_lock) {
+                SFX_lock.Select(-1);
+              } else {
+                SFX_lockup.Select(-1);
+              }
+              SFX_bgnlock.Select(-1);
+              SFX_endlock.Select(-1);
+            }
+#endif        
 #ifdef FETT263_BATTLE_MODE_ALWAYS_ON
-          battle_mode_ = true;
+            battle_mode_ = true;
 #endif
+          }
+          return true;
         }
-        return true;
-
-      case EVENTID(BUTTON_POWER, EVENT_CLICK_LONG, MODE_ON):
-        if (fusor.angle1() >  M_PI / 3) {
-          StartOrStopTrack();
-        } else {
-          SaberBase::DoForce();
-        }
-        return true;
 
       case EVENTID(BUTTON_AUX, EVENT_CLICK_SHORT, MODE_ON):
       case EVENTID(BUTTON_AUX, EVENT_DOUBLE_CLICK, MODE_ON):
-        // Avoid the base and the very tip.
-        // TODO: Make blast only appear on one blade!
-        if(swing_blast_) {
-          swing_blast_ = false;
-          if (SFX_blstend) {
-            hybrid_font.PlayCommon(&SFX_blstend);
+        if (menu_) {
```

Comment:
Split out into separate function?

Reply:
I have split the menu controls into separate functions